### PR TITLE
Tyler/187 frontend reading analytics and skip section

### PR DESCRIPTION
**Pull request by tylerttam**
## Extract Reading Analytics from Gameplay and Add Skip Section

Issue Number(s): #187.

What does this PR change and why?
Replace the `Skip Article` button with the `Skip Section` button, as well as figure out a way to store the statistics of how many passages are read by the user, and whether they skipped the section or they completed it.

### Checklist

- [x] Remove the Skip Article Button
- [x] Add a Skip Section Button and Alert popup for confirmation
- [x] Add useReadingProblems hook to generate passages and monitor statistics
- [x] Calculate the number of passages read
- [x] Move onTimeComplete function to the useReadingProblems hook
- [x] Update onTimeComplete to create a reading statistics object from the metrics

### Critical Changes / Notes

- `navigation.navigate` -> `navigation.replace`
- `useReadingProblems `generates the paragraphs and statistics rather than `ReadingMain`
- `onTimeComplete` is in `useReadingProblems `now and takes in `skipped `as a parameter
- **BUG:** `ProgressBar` temporarily pauses when you click `Next Paragraph`

### Related PRs

- N/A

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

1. Click 'Start Exercises' button
2. Complete Math section
3. Start Reading section
4. Complete Reading section OR click 'Skip Section' button
5. Check console to make sure `passagesRead` and `completed` statistics are correct

### CODE REVIEW

File: src/hooks/useReadingProblems.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export default function useReadingProblems({ navigation, route }) {
+  const [storyArray, setStoryArray] = useState(getStoryArray());
+  const [page, setPage] = useState(0);
+  const [passagesRead, setPassagesRead] = useState(0);
+  const paragraph = storyArray[page];
+
+  const incrementPassagesRead = useCallback(() => {
+    setPassagesRead((prev) => prev + 1);
+  }, []);
```
Comment (samarth52) [inline src/hooks/useReadingProblems.ts]: We do not require a new function for this atomic functionality. Can you please replace line 21 with line 11

File: src/hooks/useReadingProblems.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    incrementPassagesRead();
+  };
+
+  const onTimeComplete = useCallback(
+    (skipped) => {
+      const statistics = { passagesRead, completed: !skipped };
+      // checking that statistics are correct:
+      console.log("statistics: ", statistics);
+      console.log("completed: ", statistics.completed);
+      console.log("passagesRead: ", statistics.passagesRead);
```
Comment (samarth52) [inline src/hooks/useReadingProblems.ts]: The stats are being counted correctly. Thanks for the logs, and these lines can now be removed

File: src/hooks/useReadingProblems.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      setStoryArray(getStoryArray());
+      setPage(0);
+    } else {
+      setPage(page + 1);
+    }
+    incrementPassagesRead();
+  };
+
+  const onTimeComplete = useCallback(
+    (skipped) => {
```
Comment (samarth52) [inline src/hooks/useReadingProblems.ts]: Can you give skipped a `boolean` type. It currently has an implicit `any` type

File: src/screens/Game/ReadingMain.tsx [hunk 4]
```diff
@@ -38,39 +40,40 @@ const TOTAL_TIME = gameDescriptions.Reading.minutes * 60;
...
-    setPage(0);
+  const nextSection = () => {
+    Alert.alert(
+      "Skip Reading Section",
+      "Are you sure you want to skip the Reading section?",
+      [
+        {
+          text: "Cancel",
+          style: "cancel",
+        },
```
Comment (samarth52) [inline src/screens/Game/ReadingMain.tsx]: We would not want the timer to continue counting down while the confirmation alert is shown. Pause the timer before the alert and unpause when cancel is clicked

**Review by samarth52 — CHANGES_REQUESTED** [review]
Great work on this PR! There are a couple of minor changes to be made and I have left comments for the same.

File: src/hooks/useReadingProblems.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { useState, useCallback } from "react";
+import getStoryArray from "../assets/stories";
+
+export default function useReadingProblems({ navigation, route }) {
```
Comment (samarth52) [inline src/hooks/useReadingProblems.ts]: Add types for navigation and route similar to how it's done in `ReadingMain.tsx`

**Review by samarth52 — CHANGES_REQUESTED** [review]

**Review by samarth52 — APPROVED** [review]
lgtm 👍
